src/hardware/ad7172.rs

```diff
@@ -176,11 +176,12 @@ impl Ad7172 {
         // 5000 us delay after reset.
         delay.delay_us(5000u16);
```

Comment:
Why is it 5000 if the datasheet says 500?

Reply:
It does not seem to be sufficient sometimes. I rarely still saw register corruption in the adc if I waited for 500 us here.

Reply:
I would suspect a bug or problem somewhere else then. Like wrong SPI mode or not resetting at all (as it is now).

Reply:
True. I'll move the reset to the setup adc function in adc.

Reply:
And then see if 500 us still leads to the issue with the correct spi mode.

Reply:
Issue filed, see code.